verilog/formatting/tree_unwrapper.cc

```diff
@@ -1194,6 +1271,22 @@ void TreeUnwrapper::ReshapeTokenPartitions(
       ReshapeElseClause(node, &partition);
       break;
     }
+    case NodeEnum::kIfHeader: {
+      // Append next leaf to if header ([if (]) and treat as
+      // function/method/macro call (kAppendFittingSubPartitions)
+      if (partition.Children().size() > 1) {
+        auto& if_header = *ABSL_DIE_IF_NULL(partition.LeftmostDescendant());
+        verible::MergeLeafIntoNextLeaf(&if_header);
```

Comment:
My gut feeling here is that this Merge may be too aggressive for the general `if (...` cases.  The predicate expression may not always be a simple function/task call, it could be `if (x < y)` or `if (x(...) < y(...))`.  Pushing the `if (` into the next leaf would create an oddly shaped partition.   
Could we perhaps add a few conditions to limit this transformation to when the predicate starts "like a function/method call"?